inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -268,58 +156,60 @@ def _previously_rejected(obj, eng):
-    searcher = RecordsSearch(
-        index=index, doc_type=doc_type
-    ).params(version=True)
-
-    identifiers = []
-    for field, lookup in six.iteritems(
-            current_app.config.get("HOLDING_PEN_MATCH_MAPPING", {})):
-        # Add quotes around to make the search exact
-        identifiers += ['{0}:"{1}"'.format(field, i)
-                        for i in get_value(obj.data, lookup, [])]
-    # Search for any existing record in Holding Pen, exclude self
-    if identifiers:
-        search = searcher.query(Q('query_string',
-                                query=" OR ".join(identifiers),
-                                allow_leading_wildcard=False))
-        search_result = search.execute()
-        id_list = [int(hit.id) for hit in search_result.hits]
-        matches_excluding_self = set(id_list) - set([obj.id])
-        if matches_excluding_self:
-            obj.extra_data["holdingpen_ids"] = list(matches_excluding_self)
-            pending_records = db.session.query(
-                WorkflowObjectModel
-            ).with_entities(WorkflowObjectModel.id).filter(
-                WorkflowObjectModel.status != ObjectStatus.COMPLETED,
-                WorkflowObjectModel.id.in_(matches_excluding_self)
-            ).all()
-            if pending_records:
-                pending_ids = [o[0] for o in pending_records]
-                obj.extra_data['pending_holdingpen_ids'] = pending_ids
-                obj.log.info(
-                    "Pending records already found in Holding Pen ({0})"
-                    .format(
-                        pending_ids
-                    )
-                )
-                return True
+    """Return ``True`` if the record is already present in the Holding Pen.
+
+    Uses a custom configuration of the ``inspire-matcher`` to find duplicates
+    of the current workflow object in the Holding Pen.
+
+    Also sets ``holdingpen_matches`` in ``extra_data`` to the list of ids that
+    matched.
+
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the workflow object has a duplicate in the Holding
+        Pen or ``extra_data`` contains a true value in
+        ``already-in-holding-pen``, ``False`` otherwise.
+
+    """
+    if obj.extra_data.get('already-in-holding-pen'):
+        return True
+
+    config = {
+        'algorithm': [
+            {
+                'queries': [
+                    {
+                        'match': 'arxiv_eprints.value',
+                        'search': 'metadata.arxiv_eprints.value.raw',
+                        'type': 'exact',
+                    },
+                    {
+                        'match': 'dois.value',
+                        'search': 'metadata.dois.value.raw',
+                        'type': 'exact',
+                    },
+                ],
+            },
+        ],
+        'doc_type': 'hep',
+        'index': 'holdingpen-hep',
+    }
```

Comment:
This should live in `inspirehep/modules/workflows/config.py`, but:
1. This file does not yet exist;
2. It's not yet loaded by `inspirehep/modules/workflows/ext.py`;
3. That file is a mess. Seriously, what's up with 2629743? https://github.com/inspirehep/inspire-next/blob/f64f0eee6a5a2c0eb996352f1f10371eb64cf530/inspirehep/modules/workflows/ext.py#L53-L54